app/controllers/restaurants_controller.rb

```diff
@@ -40,4 +48,22 @@ def set_restaurant
     def restaurant_params
       params.require(:restaurant).permit(:name, :cuisine, :rating, :ten_bis, :address, :max_delivery_time)
     end
+
+  def create_where_params_obj(query_params)
+    params = {}
+
+    if query_params['rating'].present?
```

Comment:
In order to override params with default values, I suggest using the `merge` method instead.